### ISSUE TITLE
System error: read parent: connection reset by peer, fails on create, Upgrade to docker 1.10.0 when available

### ISSUE THREAD

**Issue by labboy0276** [kalabox/kalabox#904]
Hello!

Fresh install of 0.11.1 and it looks like SolR container is failing on create:

```
john@kalamuna:~/kalabox/apps$ kbox create pantheon
? Choose a Pantheon account. john@kalamuna.com
? Which site? k2m
? Which environment? dev
? What will you call this monster you have created: k2m
info: DOCKER-MACHINE =>  Starting "Kalabox2"...
info: DOCKER-MACHINE =>  (Kalabox2) Waiting for an IP...
info: DOCKER-MACHINE =>  Machine "Kalabox2" was started.
info: DOCKER-MACHINE =>  Started machines may have new IP addresses. You may need to re-run the `docker-machine env` command.
info: DOCKER-MACHINE =>  Run exited with code: 0
info: COMPOSE =>  0=--project-name, 1=kalabox, 2=--file, 3=/home/john/.kalabox/downloads/kalabox/kalabox-1.yml, 4=ps, 5=-q, 6=syncthing, 7=undefined
info: COMPOSE =>  0=--project-name, 1=kalabox, 2=--file, 3=/home/john/.kalabox/downloads/kalabox/kalabox-2.yml, 4=run, 5=--entrypoint, 6=/bin/mkdir, 7=--rm, 8=syncthing, 9=-p, 10=/code/k2m, mode=false
info: COMPOSE =>  0=--project-name, 1=kalabox, 2=--file, 3=/home/john/.kalabox/downloads/kalabox/kalabox-3.yml, 4=run, 5=--entrypoint, 6=touch, 7=--rm, 8=syncthing, 9=/code/k2m/.stfolder, mode=false
info: COMPOSE =>  0=--project-name, 1=kalabox, 2=--file, 3=/home/john/.kalabox/downloads/kalabox/kalabox-4.yml, 4=run, 5=--entrypoint, 6=/bin/cp, 7=--rm, 8=syncthing, 9=/src/kalabox/apps/k2m/code/.stignore, 10=/code/k2m/.stignore, mode=false
info: COMPOSE =>  0=--project-name, 1=kalabox, 2=--file, 3=/home/john/.kalabox/downloads/kalabox/kalabox-5.yml, 4=ps, 5=-q, 6=syncthing, 7=undefined
info: COMPOSE =>  0=--project-name, 1=k2m, 2=--file, 3=/home/john/kalabox/apps/k2m/kalabox-compose.yml, 4=--file, 5=/home/john/kalabox/apps/k2m/kalabox-cli.yml, 6=--file, 7=/home/john/.kalabox/downloads/k2m/k2m-6.yml, 8=up, 9=-d, mode=collect
Creating k2m_redis_1
Creating k2m_data_1
Creating k2m_solr_1
ERROR: Cannot start container ccfa8c2367933e5246faa3a50ab92526ec2eb43cf4d18e8d81c4420752a834f7: [8] System error: read parent: connection reset by peer
info: DOCKER-COMPOSE =>  Run exited with code: 1
code1err:more:
VError: code1err:more:
    at ChildProcess.<anonymous> (/usr/local/bin/lib/util/shell.js.jx:247:16)
    at ChildProcess.emit (events.js:85:9)
    at maybeClose (child_process.js:773:12)
    at Process.ChildProcess._handle.onexit (child_process.js:839:1)
From previous event:
    at Object.exports.spawn (/usr/local/bin/lib/util/shell.js.jx:211:10)
    at Object.sh (/usr/local/bin/plugins/kalabox-engine-docker/provider/docker/lib/bin.js.jx:92:20)
    at /usr/local/bin/plugins/kalabox-engine-docker/provider/docker/compose.js.jx:60:18
From previous event:
    at shCompose (/usr/local/bin/plugins/kalabox-engine-docker/provider/docker/compose.js.jx:42:6)
    at Object.up [as start] (/usr/local/bin/plugins/kalabox-engine-docker/provider/docker/compose.js.jx:257:12)
    at /usr/local/bin/plugins/kalabox-engine-docker/index.js.jx:129:22
From previous event:
    at Object.start (/usr/local/bin/plugins/kalabox-engine-docker/index.js.jx:128:20)
From previous event:
    at /usr/local/bin/lib/engine.js.jx:430:29
From previous event:
    at Object.exports.start (/usr/local/bin/lib/engine.js.jx:429:4)
    at /usr/local/bin/lib/app.js.jx:352:19
From previous event:
    at Object.exports.start (/usr/local/bin/lib/app.js.jx:351:4)
    at /usr/local/bin/lib/create.js.jx:404:25
From previous event:
    at Promise.try.then.then.then.then.then.then.then.tap.tap.sysRoot (/usr/local/bin/lib/create.js.jx:403:8)
    at process._tickCallback (node.js:803:1)
From previous event:
    at createApp (/usr/local/bin/lib/create.js.jx:385:6)
    at PromptUI.completed (/usr/local/bin/lib/create.js.jx:495:16)
    at PromptUI.onCompletion (/usr/local/bin/node_modules/inquirer/lib/ui/prompt.js.jx:68:10)
    at AnonymousObserver.Rx.AnonymousObserver.AnonymousObserver.completed (/usr/local/bin/node_modules/inquirer/node_modules/rx/dist/rx.js.jx:1792:12)
    at AnonymousObserver.Rx.internals.AbstractObserver.AbstractObserver.onCompleted (/usr/local/bin/node_modules/inquirer/node_modules/rx/dist/rx.js.jx:1729:14)
    at AnonymousObserver.tryCatcher (/usr/local/bin/node_modules/inquirer/node_modules/rx/dist/rx.js.jx:566:29)
    at AutoDetachObserverPrototype.completed (/usr/local/bin/node_modules/inquirer/node_modules/rx/dist/rx.js.jx:5287:56)
    at AutoDetachObserver.Rx.internals.AbstractObserver.AbstractObserver.onCompleted (/usr/local/bin/node_modules/inquirer/node_modules/rx/dist/rx.js.jx:1729:14)
    at InnerObserver.onCompleted (/usr/local/bin/node_modules/inquirer/node_modules/rx/dist/rx.js.jx:3424:65)
    at InnerObserver.tryCatcher (/usr/local/bin/node_modules/inquirer/node_modules/rx/dist/rx.js.jx:566:29)
    at AutoDetachObserverPrototype.completed (/usr/local/bin/node_modules/inquirer/node_modules/rx/dist/rx.js.jx:5287:56)
    at AutoDetachObserver.Rx.internals.AbstractObserver.AbstractObserver.onCompleted (/usr/local/bin/node_modules/inquirer/node_modules/rx/dist/rx.js.jx:1729:14)
    at AutoDetachObserver.tryCatcher (/usr/local/bin/node_modules/inquirer/node_modules/rx/dist/rx.js.jx:566:29)
    at AutoDetachObserverPrototype.completed (/usr/local/bin/node_modules/inquirer/node_modules/rx/dist/rx.js.jx:5287:56)
    at AutoDetachObserver.Rx.internals.AbstractObserver.AbstractObserver.onCompleted (/usr/local/bin/node_modules/inquirer/node_modules/rx/dist/rx.js.jx:1729:14)
    at InnerObserver.onCompleted (/usr/local/bin/node_modules/inquirer/node_modules/rx/dist/rx.js.jx:3424:65)
    at InnerObserver.tryCatcher (/usr/local/bin/node_modules/inquirer/node_modules/rx/dist/rx.js.jx:566:29)
    at AutoDetachObserverPrototype.completed (/usr/local/bin/node_modules/inquirer/node_modules/rx/dist/rx.js.jx:5287:56)
    at AutoDetachObserver.Rx.internals.AbstractObserver.AbstractObserver.onCompleted (/usr/local/bin/node_modules/inquirer/node_modules/rx/dist/rx.js.jx:1729:14)
```

**Comment by pirog** [comment]
Give the app a different name and try again.

This is a known issue that is already posted a few places. Try to check out
the open issues before posting a duplicate

On Thursday, January 28, 2016, John Ouellet <notifications@github.com>
wrote:

> Hello!
>
> Fresh install of 0.11.1 and it looks like SolR container is failing on
> create:
>
> john@kalamuna:~/kalabox/apps$ kbox create pantheon
> ? Choose a Pantheon account. john@kalamuna.com <javascript:_e(%7B%7D,'cvml','john@kalamuna.com');>
> ? Which site? k2m
> ? Which environment? dev
> ? What will you call this monster you have created: k2m
> info: DOCKER-MACHINE =>  Starting "Kalabox2"...
> info: DOCKER-MACHINE =>  (Kalabox2) Waiting for an IP...
> info: DOCKER-MACHINE =>  Machine "Kalabox2" was started.
> info: DOCKER-MACHINE =>  Started machines may have new IP addresses. You may need to re-run the `docker-machine env` command.
> info: DOCKER-MACHINE =>  Run exited with code: 0
> info: COMPOSE =>  0=--project-name, 1=kalabox, 2=--file, 3=/home/john/.kalabox/downloads/kalabox/kalabox-1.yml, 4=ps, 5=-q, 6=syncthing, 7=undefined
> info: COMPOSE =>  0=--project-name, 1=kalabox, 2=--file, 3=/home/john/.kalabox/downloads/kalabox/kalabox-2.yml, 4=run, 5=--entrypoint, 6=/bin/mkdir, 7=--rm, 8=syncthing, 9=-p, 10=/code/k2m, mode=false
> info: COMPOSE =>  0=--project-name, 1=kalabox, 2=--file, 3=/home/john/.kalabox/downloads/kalabox/kalabox-3.yml, 4=run, 5=--entrypoint, 6=touch, 7=--rm, 8=syncthing, 9=/code/k2m/.stfolder, mode=false
> info: COMPOSE =>  0=--project-name, 1=kalabox, 2=--file, 3=/home/john/.kalabox/downloads/kalabox/kalabox-4.yml, 4=run, 5=--entrypoint, 6=/bin/cp, 7=--rm, 8=syncthing, 9=/src/kalabox/apps/k2m/code/.stignore, 10=/code/k2m/.stignore, mode=false
> info: COMPOSE =>  0=--project-name, 1=kalabox, 2=--file, 3=/home/john/.kalabox/downloads/kalabox/kalabox-5.yml, 4=ps, 5=-q, 6=syncthing, 7=undefined
> info: COMPOSE =>  0=--project-name, 1=k2m, 2=--file, 3=/home/john/kalabox/apps/k2m/kalabox-compose.yml, 4=--file, 5=/home/john/kalabox/apps/k2m/kalabox-cli.yml, 6=--file, 7=/home/john/.kalabox/downloads/k2m/k2m-6.yml, 8=up, 9=-d, mode=collect
> Creating k2m_redis_1
> Creating k2m_data_1
> Creating k2m_solr_1
> ERROR: Cannot start container ccfa8c2367933e5246faa3a50ab92526ec2eb43cf4d18e8d81c4420752a834f7: [8] System error: read parent: connection reset by peer
> info: DOCKER-COMPOSE =>  Run exited with code: 1
> code1err:more:
> VError: code1err:more:
>     at ChildProcess.<anonymous> (/usr/local/bin/lib/util/shell.js.jx:247:16)
>     at ChildProcess.emit (events.js:85:9)
>     at maybeClose (child_process.js:773:12)
>     at Process.ChildProcess._handle.onexit (child_process.js:839:1)
> From previous event:
>     at Object.exports.spawn (/usr/local/bin/lib/util/shell.js.jx:211:10)
>     at Object.sh (/usr/local/bin/plugins/kalabox-engine-docker/provider/docker/lib/bin.js.jx:92:20)
>     at /usr/local/bin/plugins/kalabox-engine-docker/provider/docker/compose.js.jx:60:18
> From previous event:
>     at shCompose (/usr/local/bin/plugins/kalabox-engine-docker/provider/docker/compose.js.jx:42:6)
>     at Object.up [as start] (/usr/local/bin/plugins/kalabox-engine-docker/provider/docker/compose.js.jx:257:12)
>     at /usr/local/bin/plugins/kalabox-engine-docker/index.js.jx:129:22
> From previous event:
>     at Object.start (/usr/local/bin/plugins/kalabox-engine-docker/index.js.jx:128:20)
> From previous event:
>     at /usr/local/bin/lib/engine.js.jx:430:29
> From previous event:
>     at Object.exports.start (/usr/local/bin/lib/engine.js.jx:429:4)
>     at /usr/local/bin/lib/app.js.jx:352:19
> From previous event:
>     at Object.exports.start (/usr/local/bin/lib/app.js.jx:351:4)
>     at /usr/local/bin/lib/create.js.jx:404:25
> From previous event:
>     at Promise.try.then.then.then.then.then.then.then.tap.tap.sysRoot (/usr/local/bin/lib/create.js.jx:403:8)
>     at process._tickCallback (node.js:803:1)
> From previous event:
>     at createApp (/usr/local/bin/lib/create.js.jx:385:6)
>     at PromptUI.completed (/usr/local/bin/lib/create.js.jx:495:16)
>     at PromptUI.onCompletion (/usr/local/bin/node_modules/inquirer/lib/ui/prompt.js.jx:68:10)
>     at AnonymousObserver.Rx.AnonymousObserver.AnonymousObserver.completed (/usr/local/bin/node_modules/inquirer/node_modules/rx/dist/rx.js.jx:1792:12)
>     at AnonymousObserver.Rx.internals.AbstractObserver.AbstractObserver.onCompleted (/usr/local/bin/node_modules/inquirer/node_modules/rx/dist/rx.js.jx:1729:14)
>     at AnonymousObserver.tryCatcher (/usr/local/bin/node_modules/inquirer/node_modules/rx/dist/rx.js.jx:566:29)
>     at AutoDetachObserverPrototype.completed (/usr/local/bin/node_modules/inquirer/node_modules/rx/dist/rx.js.jx:5287:56)
>     at AutoDetachObserver.Rx.internals.AbstractObserver.AbstractObserver.onCompleted (/usr/local/bin/node_modules/inquirer/node_modules/rx/dist/rx.js.jx:1729:14)
>     at InnerObserver.onCompleted (/usr/local/bin/node_modules/inquirer/node_modules/rx/dist/rx.js.jx:3424:65)
>     at InnerObserver.tryCatcher (/usr/local/bin/node_modules/inquirer/node_modules/rx/dist/rx.js.jx:566:29)
>     at AutoDetachObserverPrototype.completed (/usr/local/bin/node_modules/inquirer/node_modules/rx/dist/rx.js.jx:5287:56)
>     at AutoDetachObserver.Rx.internals.AbstractObserver.AbstractObserver.onCompleted (/usr/local/bin/node_modules/inquirer/node_modules/rx/dist/rx.js.jx:1729:14)
>     at AutoDetachObserver.tryCatcher (/usr/local/bin/node_modules/inquirer/node_modules/rx/dist/rx.js.jx:566:29)
>     at AutoDetachObserverPrototype.completed (/usr/local/bin/node_modules/inquirer/node_modules/rx/dist/rx.js.jx:5287:56)
>     at AutoDetachObserver.Rx.internals.AbstractObserver.AbstractObserver.onCompleted (/usr/local/bin/node_modules/inquirer/node_modules/rx/dist/rx.js.jx:1729:14)
>     at InnerObserver.onCompleted (/usr/local/bin/node_modules/inquirer/node_modules/rx/dist/rx.js.jx:3424:65)
>     at InnerObserver.tryCatcher (/usr/local/bin/node_modules/inquirer/node_modules/rx/dist/rx.js.jx:566:29)
>     at AutoDetachObserverPrototype.completed (/usr/local/bin/node_modules/inquirer/node_modules/rx/dist/rx.js.jx:5287:56)
>     at AutoDetachObserver.Rx.internals.AbstractObserver.AbstractObserver.onCompleted (/usr/local/bin/node_modules/inquirer/node_modules/rx/dist/rx.js.jx:1729:14)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kalabox/kalabox/issues/904>.
>


-- 
Mike Pirog
Co-Founder, Kalabox Inc

**Comment by labboy0276** [comment]
Yes sorry, I just saw similar to here:  https://github.com/kalabox/kalabox/issues/894

**Comment by pirog** [comment]
New app bane needs to have a different amount of characters.

This is a well known docker bug so I'd check out their issue queues as well

**Comment by labboy0276** [comment]
“Calm down Doctor, now is not the time for fear. That comes later.”

**Comment by labboy0276** [comment]
There are work arounds and a resolution at the end of this long ticket if need be:
https://github.com/docker/docker/issues/14203

**Comment by labboy0276** [comment]
Also,

I had a ton of old kalabox items in ~/.npm which i removed

changing the name of the app seems to resolve the issue.

**Comment by labboy0276** [comment]
Another workaround (if you don;t want to rename your app), add this to the failing container in your kalabox-compose.yml:

```
  environment:
    DOCKER_FIX: '                                        '
```

**Comment by labboy0276** [comment]
And finally, it seems this will be fixed in docker 1.9.2 once that rolls out.

**Comment by pirog** [comment]
Just to consolidate all the comments so far:

This issue is a lower level docker issue that is tracked over here

**Comment by pirog** [comment]
this should be resolved when we roll out 0.11.3
